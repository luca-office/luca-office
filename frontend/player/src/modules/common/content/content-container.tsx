import * as React from "react"
import {useDispatch, useSelector} from "react-redux"
import {usePlayerWebsocket} from "../../../hooks/use-player-websocket"
import {navigateToRouteAction} from "../../../redux/actions/routing-action"
import {AppState} from "../../../redux/state/app-state"
import {Route, RouteParameters} from "../../../routes"
import {Content} from "./content"

export const ContentContainer: React.FC = () => {
  const activeRoute = useSelector<AppState, Route | undefined>(store => store.routing.activeRoute)
  const routingParameters = useSelector<AppState, RouteParameters | undefined>(store => store.routing.parameters)
  const isChatOpened = useSelector<AppState, boolean>(store => store.ui.windowManager.isChatVisible)
  const dispatch = useDispatch()

  usePlayerWebsocket(isChatOpened)

  const handleCloseScenarioFinalScoreOverlay = () => {
    dispatch(
      navigateToRouteAction({
        routeType: Route.ReportScenario,
        parameters: {
          scenarioId: routingParameters?.scenarioId,
          surveyId: routingParameters?.surveyId,
          token: routingParameters?.token
        }
      })
    )
  }
  const handleCloseQuestionnaireFinalScoreOverlay = () => {
    dispatch(
      navigateToRouteAction({
        routeType: Route.ReportQuestionnaire,
        parameters: {
          questionnaireId: routingParameters?.questionnaireId,
          surveyId: routingParameters?.surveyId,
          token: routingParameters?.token
        }
      })
    )
  }

  return (
    <Content
      closeScenarioFinalScoreOverlay={handleCloseScenarioFinalScoreOverlay}
      closeQuestionnaireFinalScoreOverlay={handleCloseQuestionnaireFinalScoreOverlay}
      routingParameters={routingParameters ?? {}}
      activeRoute={activeRoute}
    />
  )
}
