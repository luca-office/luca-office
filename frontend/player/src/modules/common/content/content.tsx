import * as React from "react"
import {Option} from "shared/utils"
import {ReportingLoginContainer} from "../../auth/reporting-login/reporting-login-container"
import {Route, RouteParameters} from "../../../routes"
import {VerifyTokenContainer} from "../../auth"
import {Login} from "../../auth/login/login"
import {ProjectResumptionContainer} from "../../auth/login/project-resumption/project-resumption-container"
import {OpenParticipationResumptionOrStartContainer} from "../../auth/open-participation/open-participation-resumption-or-start-container"
import {ModuleQuestionnaire} from "../../questionnaire/module-questionnaire/module-questionnaire"
import {ReportingContentContainer} from "../../reporting/reporting-content-container/reporting-content-container"
import {ReportingQuestionnaireOverlayContainer} from "../../reporting/reporting-questionnaire-overlay/reporting-questionnaire-overlay-container"
import {ReportingScenarioOverlayContainer} from "../../reporting/reporting-scenario-overlay/reporting-scenario-overlay-container"
import {DesktopContainer} from "../desktop/desktop-container"
import {ProjectEndContainer} from "../project-end/project-end-container"
import {WaitForNextModule} from "../wait-for-next-module/wait-for-next-module"
import {WelcomeModalContainer} from "../welcome-modal/welcome-modal-container"

interface ContentProps {
  readonly routingParameters?: RouteParameters
  readonly activeRoute: Route | undefined
  readonly closeScenarioFinalScoreOverlay: () => void
  readonly closeQuestionnaireFinalScoreOverlay: () => void
}

export const Content: React.FC<ContentProps> = ({
  routingParameters,
  activeRoute,
  closeScenarioFinalScoreOverlay,
  closeQuestionnaireFinalScoreOverlay
}) => {
  switch (activeRoute) {
    case Route.OpenParticipation:
      return <OpenParticipationResumptionOrStartContainer surveyId={Option.of(routingParameters?.surveyId)} />
    case Route.VerifyToken:
      return <VerifyTokenContainer />
    case Route.Login:
      return <Login originRoute={Route.Login} />
    case Route.StartProject:
      return <Login originRoute={Route.StartProject} />
    case Route.ResumeProject:
      return <ProjectResumptionContainer />
    case Route.Welcome:
      return <WelcomeModalContainer firstProjectModule={routingParameters?.firstProjectModule} />
    case Route.ScenarioDetail:
      return <DesktopContainer scenarioId={Option.of(routingParameters?.scenarioId)} />
    case Route.Questionnaire:
      return <ModuleQuestionnaire questionnaireIdOption={Option.of(routingParameters?.questionnaireId)} />
    case Route.EndOfProject:
      return <ProjectEndContainer />
    case Route.WaitForNextModule:
      return <WaitForNextModule isStartScreenOfManualSurvey={false} />
    case Route.WaitForManualSurveyStart:
      return <WaitForNextModule isStartScreenOfManualSurvey={true} />
    case Route.ReportLogin:
      return <ReportingLoginContainer surveyId={routingParameters?.surveyId} token={routingParameters?.token} />
    case Route.Report:
      return (
        <ReportingContentContainer
          surveyId={routingParameters?.surveyId}
          token={routingParameters?.token}
          scenarioId={null}
          questionnaireId={null}
        />
      )
    case Route.ReportScenario:
    case Route.ReportScenarioScoringOverlay:
      return (
        <React.Fragment>
          <ReportingContentContainer
            surveyId={routingParameters?.surveyId}
            token={routingParameters?.token}
            scenarioId={routingParameters?.scenarioId}
            questionnaireId={null}
          />
          {activeRoute === Route.ReportScenarioScoringOverlay && (
            <ReportingScenarioOverlayContainer
              onCloseOverlay={closeScenarioFinalScoreOverlay}
              surveyId={routingParameters?.surveyId}
              surveyInvitationId={routingParameters?.surveyInvitationId}
              scenarioId={routingParameters?.scenarioId}
              participantName={routingParameters?.participantName}
            />
          )}
        </React.Fragment>
      )
    case Route.ReportQuestionnaire:
    case Route.ReportQuestionnaireScoringOverlay:
      return (
        <React.Fragment>
          <ReportingContentContainer
            surveyId={routingParameters?.surveyId}
            token={routingParameters?.token}
            scenarioId={null}
            questionnaireId={routingParameters?.questionnaireId}
          />
          {activeRoute === Route.ReportQuestionnaireScoringOverlay && (
            <ReportingQuestionnaireOverlayContainer
              surveyId={routingParameters?.surveyId}
              onCloseOverlay={closeQuestionnaireFinalScoreOverlay}
              surveyInvitationId={routingParameters?.surveyInvitationId}
              questionnaireId={routingParameters?.questionnaireId}
              participantName={routingParameters?.participantName}
            />
          )}
        </React.Fragment>
      )
    default:
      return <VerifyTokenContainer />
  }
}
