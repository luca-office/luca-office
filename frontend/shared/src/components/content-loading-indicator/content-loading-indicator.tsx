import {css} from "@emotion/react"
import {CSSInterpolation} from "@emotion/serialize"
import * as React from "react"
import {CustomStyle, Flex, spacingMedium} from "../../styles"
import {LoadingIndicator} from "../loading-indicator/loading-indicator"

export interface ContentLoadingIndicatorProps extends CustomStyle {
  readonly customLoadingIndicatorStyles?: CSSInterpolation
}

export const ContentLoadingIndicator: React.FC<ContentLoadingIndicatorProps> = ({
  customStyles,
  customLoadingIndicatorStyles
}) => (
  <div css={[styles.placeholderGeneral, customStyles]}>
    <LoadingIndicator {...{...(!!customLoadingIndicatorStyles && {customStyles: customLoadingIndicatorStyles})}} />
  </div>
)

const styles = {
  placeholderGeneral: css(Flex.row, {
    padding: spacingMedium,
    justifyContent: "center"
  })
}
