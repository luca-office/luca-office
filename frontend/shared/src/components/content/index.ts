export * from "./content/content"
export * from "./content-bottom-actionbar/content-bottom-actionbar"
export * from "./content-missing-indicator/content-missing-indicator"
