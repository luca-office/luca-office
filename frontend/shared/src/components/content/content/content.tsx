import {css} from "@emotion/react"
import {CSSInterpolation} from "@emotion/serialize"
import * as React from "react"
import {PropsWithChildren} from "react"
import {CustomStyle, Flex, headerHeight, spacing, spacingHuge, spacingMedium} from "../../../styles"
import {ContentLoadingIndicator} from "../../content-loading-indicator/content-loading-indicator"
import {ContentBottomActionbar, ContentMissingIndicator} from "../index"

interface Props extends CustomStyle {
  readonly actionBar?: React.ReactNode
  readonly customContentContainerStyles?: CSSInterpolation
  readonly customFooterStyles?: CSSInterpolation
  readonly customLoadingIndicatorStyles?: CSSInterpolation
  readonly isContentMissing?: boolean
  readonly loading?: boolean
  readonly subHeader?: React.ReactNode
}

export const Content: React.FunctionComponent<PropsWithChildren<Props>> = ({
  actionBar,
  children,
  customContentContainerStyles,
  customFooterStyles,
  customLoadingIndicatorStyles,
  customStyles,
  isContentMissing,
  loading,
  subHeader
}) => {
  return (
    <div className="content" css={[styles.contentWrapper, customStyles]}>
      {subHeader}
      {loading && <ContentLoadingIndicator customStyles={customLoadingIndicatorStyles} />}
      {!loading && isContentMissing && (
        <ContentMissingIndicator customLabelledCardStyles={styles.contentMissingIndicator} />
      )}
      {!loading && !isContentMissing && (
        <React.Fragment>
          <div className="content-container" css={[styles.content, customContentContainerStyles]}>
            {children}
          </div>
          {actionBar && <ContentBottomActionbar customStyles={customFooterStyles}>{actionBar}</ContentBottomActionbar>}
        </React.Fragment>
      )}
    </div>
  )
}

const styles = {
  contentWrapper: css(Flex.column, {
    flexGrow: 1,
    minHeight: `calc(100vh - ${headerHeight}px)`
  }),
  content: css({
    flex: 1,
    padding: spacing(spacingMedium, spacingHuge)
  }),
  contentMissingIndicator: css(Flex.column, {
    justifyContent: "center",
    alignItems: "center"
  })
}
