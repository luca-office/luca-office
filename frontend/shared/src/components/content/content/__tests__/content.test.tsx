import {mount} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {ContentLoadingIndicator} from "../../../content-loading-indicator/content-loading-indicator"
import {ContentBottomActionbar, ContentMissingIndicator} from "../../index"
import {Content} from "../content"

describe("content", () => {
  it("renders correctly", () => {
    const component = (
      <Content>
        <div>test content</div>
      </Content>
    )
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correct structure", () => {
    const component = (
      <Content subHeader={<span>header</span>} actionBar={"actions"}>
        <div className="test">test content</div>
      </Content>
    )
    const tree = mount(component)
    const html = tree.html()

    expect(tree.find(".test")).toHaveLength(1)
    expect(tree.find("div")).toHaveLength(4)
    expect(tree.find("span")).toHaveLength(1)
    expect(tree.find(ContentBottomActionbar)).toHaveLength(1)
    expect(html).toContain("test content")
    expect(html).toContain("header")
    expect(html).toContain("actions")
  })
  it("renders correct structure loading", () => {
    const component = (
      <Content subHeader={<span>header</span>} actionBar={"actions"} loading={true}>
        <div className="test">test content</div>
      </Content>
    )
    const tree = mount(component)

    expect(tree.find(".test")).toHaveLength(0)
    expect(tree.find("div")).toHaveLength(3)
    expect(tree.find("span")).toHaveLength(1)
    expect(tree.find(ContentLoadingIndicator)).toHaveLength(1)
    expect(tree.find(ContentBottomActionbar)).toHaveLength(0)
  })
  it("renders correct structure missing content", () => {
    const component = (
      <Content subHeader={<span>header</span>} actionBar={"actions"} isContentMissing={true}>
        <div className="test">test content</div>
      </Content>
    )
    const tree = mount(component)

    expect(tree.find(".test")).toHaveLength(0)
    expect(tree.find("span")).toHaveLength(1)
    expect(tree.find(ContentMissingIndicator)).toHaveLength(1)
    expect(tree.find(ContentBottomActionbar)).toHaveLength(0)
  })
  it("renders correct structure missing content and loading", () => {
    const component = (
      <Content subHeader={<span>header</span>} actionBar={"actions"} isContentMissing={true} loading={true}>
        <div className="test">test content</div>
      </Content>
    )
    const tree = mount(component)

    expect(tree.find(".test")).toHaveLength(0)
    expect(tree.find("div")).toHaveLength(3)
    expect(tree.find("span")).toHaveLength(1)
    expect(tree.find(ContentMissingIndicator)).toHaveLength(0)
    expect(tree.find(ContentLoadingIndicator)).toHaveLength(1)
    expect(tree.find(ContentBottomActionbar)).toHaveLength(0)
  })
})
