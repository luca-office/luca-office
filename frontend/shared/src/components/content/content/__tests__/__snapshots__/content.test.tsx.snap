// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content renders correctly 1`] = `
<div
  className="content css-1m6jcw1-Content"
>
  <div
    className="content-container css-ckgebi-Content"
  >
    <div>
      test content
    </div>
  </div>
</div>
`;
